### PR TITLE
Use the status from the item_display field to determine on-order-ness

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

File: lib/holdings/item.rb [hunk 1]
```diff
@@ -34,7 +33,8 @@ def browsable?
     end
 
     def on_order?
-      current_location.code == 'ON-ORDER'
+      folio_status == 'On order' ||
+        current_location.code == 'ON-ORDER'
```
Comment (jcoyne) [inline lib/holdings/item.rb:37]: Is there a reason we still need to support the location code?

Comment (cbeer) [reply lib/holdings/item.rb:37]: At least until we do a full re-index with https://github.com/sul-dlss/searchworks_traject_indexer/pull/1275, on-order items without a FOLIO item don't have the status yet.

Comment (jcoyne) [reply lib/holdings/item.rb:37]: Are we going to be able to remember that? Should we drop a comment here or create a new ticket that points to that?

**Review by jcoyne — APPROVED** [review]
Looks good. Feel free to merge after considering a comment or ticket for tracking the removal of the old method.

**Pull request merged**